src/highdicom/content.py

```diff
@@ -1478,15 +1503,318 @@ def __init__(
                 )
 
 
+class ColorLUT(Dataset):
```

Comment:
The *highdicom* interface (protocol) is very similar indeed, but the attributes have very different names (`RedPaletteColorLookupTableDescriptor` instead of `LUTDescriptor`, etc.)

Reply:
Yeah, it's a mess. I suppose the duplicated code is quite simple so no big deal.

It's just quite funny because my initial implementation split duplicated this logic between PaletteColorLUT and LUT  and then you asked me to merge it to reduce duplication. And now you've duplicated it again >_<

Reply:
Whatever you think is best I'll go with

Reply:
> It's just quite funny because my initial implementation split duplicated this logic between PaletteColorLUT and LUT and then you asked me to merge it to reduce duplication (though there were other issues). And now you've duplicated it again >_<

Yeah, I didn't appreciate the complexity initially..